### PR TITLE
Add detail to scopes_required, audience_required, groups_required and roles_required

### PR DESCRIPTION
**Pull request by liyangau**
### Summary
Add detail explanation to `scopes_required`, `audience_required`, `groups_required` and `roles_required` which are available on plugin doc.

### CODE REVIEW

File: app/_hub/kong-inc/openid-connect/index.md [hunk 1]
```diff
@@ -429,7 +429,10 @@ params:
       required: false
       default: (discovered issuer)
       datatype: array of string elements
-      description: The scopes required to be in the access token.
+      description: |
+        The scopes (`scopes_claim` claim) required to be present in the access token (or introspection results) for successful authorization. This config parameter works in both **AND** / **OR** cases. 
+        - When `["scope1 scope2"]` are in the same array indices, both `scope1` AND `scope2` need to be present in access token (or introspection results). 
+        - When `["scope1,scope2"]` are in different array indices, either `scope1` OR `scope2` need to be present in access token (or introspection results).
```
Comment (bungle) [inline app/_hub/kong-inc/openid-connect/index.md:435]:
```suggestion
        - When `["scope1", "scope2"]` are in different array indices, either `scope1` OR `scope2` need to be present in access token (or introspection results).
```

File: app/_hub/kong-inc/openid-connect/index.md [hunk 2]
```diff
@@ -439,7 +442,10 @@ params:
       required: false
       default: 
       datatype: array of string elements
-      description: The audience required to be in the access token.
+      description: |
+        The audiences (`audience_claim` claim) required to be present in the access token (or introspection results) for successful authorization. This config parameter works in both **AND** / **OR** cases. 
+        - When `["audience1 audience2"]` are in the same array indices, both `audience1` AND `audience2` need to be present in access token (or introspection results). 
+        - When `["audience1,audience2"]` are in different array indices, either `audience1` OR `audience2` need to be present in access token (or introspection results).
```
Comment (bungle) [inline app/_hub/kong-inc/openid-connect/index.md:448]:
```suggestion
        - When `["audience1", "audience2"]` are in different array indices, either `audience1` OR `audience2` need to be present in access token (or introspection results).
```

File: app/_hub/kong-inc/openid-connect/index.md [hunk 3]
```diff
@@ -449,7 +455,10 @@ params:
       required: false
       default: 
       datatype: array of string elements
-      description: The groups required to be in the access token.
+      description: |
+        The groups (`groups_claim` claim) required to be present in the access token (or introspection results) for successful authorization. This config parameter works in both **AND** / **OR** cases. 
+        - When `["group1 group2"]` are in the same array indices, both `group1` AND `group2` need to be present in access token (or introspection results). 
+        - When `["group1,group2"]` are in different array indices, either `group1` OR `group2` need to be present in access token (or introspection results).
```
Comment (bungle) [inline app/_hub/kong-inc/openid-connect/index.md:461]:
```suggestion
        - When `["group1", "group2"]` are in different array indices, either `group1` OR `group2` need to be present in access token (or introspection results).
```

File: app/_hub/kong-inc/openid-connect/index.md [hunk 4]
```diff
@@ -459,7 +468,10 @@ params:
       required: false
       default: 
       datatype: array of string elements
-      description: The roles required to be in the access token.
+      description: |
+        The roles (`roles_claim` claim) required to be present in the access token (or introspection results) for successful authorization. This config parameter works in both **AND** / **OR** cases. 
+        - When `["role1 role2"]` are in the same array indices, both `role1` AND `role2` need to be present in access token (or introspection results). 
+        - When `["role1,role2"]` are in different array indices, either `role1` OR `role2` need to be present in access token (or introspection results).
```
Comment (bungle) [inline app/_hub/kong-inc/openid-connect/index.md:474]:
```suggestion
        - When `["role1", "role2"]` are in different array indices, either `role1` OR `role2` need to be present in access token (or introspection results).
```

**Review by bungle — DISMISSED** [review]

**Review by lena-larionova — APPROVED** [review]
LGTM

**Comment by lena-larionova** [comment]
@bungle can you take a quick look and let Li know if the changes are correct?

**Pull request merged**